src/compiler/checker.ts

```diff
@@ -8223,6 +8459,60 @@ module ts {
             }
         }
 
+        function getPromiseConstructor(node: SignatureDeclaration): EntityName {
+            if (node.type && node.type.kind === SyntaxKind.TypeReference) {
+                var typeReference = <TypeReferenceNode>node.type;
+                return typeReference.typeName;
+            }
+
+            if (globalPromiseSymbol && globalPromiseSymbol === resolveName(node, "Promise", SymbolFlags.Value, undefined, undefined)) {

```

Comment:
I perform this check in `checkCollisionsWithAwaiterVaraiablesInGeneratedCode`.

From: Jason Freeman [mailto:notifications@github.com]
Sent: Friday, February 13, 2015 5:54 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/checker.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24710639>:

> @@ -8223,6 +8459,60 @@ module ts {

>              }

>          }

>

> +        function getPromiseConstructor(node: SignatureDeclaration): EntityName {

> +            if (node.type && node.type.kind === SyntaxKind.TypeReference) {

> +                var typeReference = <TypeReferenceNode>node.type;

> +                return typeReference.typeName;

> +            }

> +

> +            if (globalPromiseSymbol && globalPromiseSymbol === resolveName(node, "Promise", SymbolFlags.Value, undefined, undefined)) {

Make sure to resolveName inside the body of the function, in case there is something inside called Promise.

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24710639>.